manifests/php.pp

```diff
@@ -67,15 +61,46 @@
     $env = delete($env_temp, nil)
 
     $install_option = $ensure ? { absent => 'purge', default => 'install' }
-    exec { 'newrelic-install':
-        environment => $env,
-        command => "newrelic-install ${install_option}",
-        path => ['/bin', '/usr/bin'],
-        require => Package['newrelic-php5'],
-        refreshonly => true
+
+
+
+    if ($ensure == absent) {
+      Exec['newrelic-install'] -> Package['newrelic-php5']
+
+      if ($php_service) {
+        Package['newrelic-php5'] ~> Service[$php_service]
+      }
+
+      exec { 'newrelic-install':
+          environment => $env,
+          command     => "newrelic-install ${install_option}",
+          path        => ['/bin', '/usr/bin'],
+          onlyif      => 'test -d /usr/lib/newrelic-php5/agent/x64',
```

Comment:
This is going to need to be made more generic - this line was added as a way of preventing the Exec from running if the PHP agent has already been uninstalled, otherwise the dependency causes it to be invoked on every run.